<?xml version="1.0"?>
<launch>

    <!-- <group ns="car" clear_params="true"> -->
    <rosparam>
        use_sim_time: true
    </rosparam>
    <arg name="car_bl" value="car/base_link"/>
    <!-- <node pkg="hokuyo_node" type="hokuyo_node" name="hokuyo_node"> -->
    <!--     <param name="frame_id" value="$(arg car_bl)"/> -->
    <!-- </node> -->
    <node pkg="rosbag" type="play" name="data_player"
        args="--quiet --clock /home/wallar/storage/bags/2016-05-26-15-11-41.bag"/>
    <node pkg="ira_laser_tools" name="laserscan_multi_merger"
        type="laserscan_multi_merger" output="screen">
        <param name="destination_frame" value="front_lidar"/>
        <param name="cloud_destination_topic" value="/merged_cloud"/>
        <param name="scan_destination_topic" value="/scan_multi"/>
        <rosparam>
            laserscan_topics:
                - /golfcartdj/front_scan_changed_range
                <!-- &#45; /scan -->
        </rosparam>
    </node>
    <!-- </group> -->

    <include 
        file="$(find foresight)/launch/localization/static_transformations.launch"/>

    <arg name="set_pos" 
        value="'{header: {frame_id: 'front_lidar'},
        pose: {position: {z: 1}, orientation: {w: 1}}}'"/>
    <arg name="msg_type" value="geometry_msgs/PoseStamped"/>
    <arg name="topic_name" value="/mavros/local_position/pose"/>
    <node pkg="rostopic" type="rostopic" name="setter"
        args="pub $(arg topic_name) $(arg msg_type) $(arg set_pos) -r 30"/>

    <node pkg="tf" type="static_transform_publisher" name="map_to_quad"
        args="0 0 1 0 0 0 front_lidar base_link 100" />

    <!-- <node pkg="tf" type="static_transform_publisher" name="map_to_car" -->
    <!--     args="0 0 0 0 0 0 map car/base_link 100" /> -->
    <!-- <node pkg="foresight" type="tf_to_pose.py" name="tf_to_pose"> -->
    <!--     <param name="fixed_frame" value="golfcartdj/odom"/> -->
    <!--     <param name="child_frame" value="golfcartdj/base_link"/> -->
    <!--     <param name="pose_topic" value="/mavros/local_position/pose"/> -->
    <!-- </node> -->

    <node pkg="foresight" type="frontier_polygons.py" name="frontier_pub"
        output="screen">
        <param name="map_frame" value="front_lidar"/>
        <param name="frequency" value="30"/>
        <param name="scan_break_thresh" value="1"/>
    </node>


    <node pkg="foresight" type="info_planner.py" name="planner"
        output="screen">
        <!-- Parameters for calculating projection -->
        <param name="fov_v" value="1.05"/>
        <param name="fov_h" value="1.05"/>
        <param name="map_frame" value="front_lidar"/>
        <param name="quad_frame" value="base_link"/>
        <param name="camera_frame" value="back_camera_link"/>
        <param name="altitude" value="3"/>

        <!-- Parameters for planner -->
        <param name="optimality_threshold" value="1.0"/>
        <param name="max_execution_time" value="10"/>
        <param name="max_speed" value="4.0"/>
        <param name="neighbour_dist" value="3"/>
        <param name="timeout" value="0.2"/>

        <!-- Configuration parameters -->
        <param name="frequency" value="100"/>
        <remap from="/scan_polygon" to="/scan_polygon"/>
    </node>
</launch>
